core/domain/suggestion_services.py

```diff
@@ -144,6 +145,27 @@ def get_suggestion_by_id(suggestion_id):
     return get_suggestion_from_model(model) if model else None
 
 
+def get_suggestions_by_ids(suggestion_ids):
```

Comment:
It seems like a bunch of the code changes in this PR lead to this method still being called in a few places. Should we be leveraging caching to avoid needing to call into NDB a bunch of times when fetching suggestions?